src/mpeg4/mpeg4Tag.ts

```diff
@@ -0,0 +1,30 @@
+import CombinedTag from "../combinedTag";
+import { Tag, TagTypes } from "../tag";
+
+export default class Mpeg4Tag extends CombinedTag {
```

Comment:
In fact, this class should not exist. I see that the setTags() function is part of the CombinedTag class in the original implementation. So, logically, it should be moved to that location. Then, Mpeg4Tag would just becomes a helper to create a CombinedTag object in class Mpeg4File.read(), because the constructor of CombinedTag is protected and there is no way to access it in Mpeg4File.read(). That constructor is public in the original implementation though. Would it make sense to remove class Mpeg4Tag and make the CombinedTag constructor public? I'd like your thoughts on this.

Reply:
Ok, I took a look at the C# code and I see what you mean. This is one place where I've made some significant changes to the original implemntation. The idea for `CombinedTag` is that it wraps around multiple `Tag` instances that a file may or may not contain. This mainly exists for the benefit of the unified tagging interface so that there's only one logical tag object per file. Functionality that's special to a file type can be overridden (eg, flac gets its pictures from the `FlacPictures` tag instance, while everything else comes from `XiphComment`). If the tag types that are wrapped are not optional, then `createTag` should throw, `removeTag` should do nothing (eg, `OggTag` that wraps multiple `XiphComments` that exist on each stream).

This is a different situation than usual, though. The only reason the C# implementation opted to create `Mpeg4Tag` was because both apple and id3v2 tags should be supported. But, as we've established, Id3v2 tags aren't implemented yet! So I think there's two choices:

1) (my preferred) Remove this class just set `Mpeg4File.tag` to the `AppleTag` instance. If we ever get around to implementing ID3v2 tag support, we'll reinstate `Mpeg4Tag` and have it actually wrap the two tags.
2) Follow the pattern that `OggTag` sets with `createTag` throwing and `removeTags` being a no-op (although I realize now this violates SOLID principles).

I'll leave it up to you, but I think option is going to be the least amount of work 😄

Reply:
I've chosen for option 1 in 5cce6c01550973c99265d6fe9a75afcd752e8f43. Could you have a look if that's what you had in mind?

Reply:
Yep, exactly what I had in mind! 👍